src/mgmt/cli_mgmt.c

```diff
@@ -625,14 +641,52 @@ dc_mgmt_net_cfg(const char *name, crt_init_options_t *crt_info)
 	D_STRNDUP(crt_info->cio_provider, info->provider, DAOS_SYS_INFO_STRING_MAX);
 	if (NULL == crt_info->cio_provider)
 		D_GOTO(cleanup, rc = -DER_NOMEM);
-	D_STRNDUP(crt_info->cio_interface, info->interface, DAOS_SYS_INFO_STRING_MAX);
-	if (NULL == crt_info->cio_interface)
-		D_GOTO(cleanup, rc = -DER_NOMEM);
-	D_STRNDUP(crt_info->cio_domain, info->domain, DAOS_SYS_INFO_STRING_MAX);
-	if (NULL == crt_info->cio_domain)
-		D_GOTO(cleanup, rc = -DER_NOMEM);
 
-	D_INFO("Network interface: %s, Domain: %s\n", info->interface, info->domain);
+	d_getenv_bool("D_DYNAMIC_CTX", &d_dynamic_ctx_g);
+	if (d_dynamic_ctx_g) {
+		int i, cnt = 0;
+
+		for (i = 0; i < resp->n_numa_fabric_interfaces; i++)
+			cnt += resp_g->numa_fabric_interfaces[i]->n_ifaces + 1;
```

Comment:
is it intended to add +1 for each repetition of loop rather than at the end once? I don't quite understand what +1 is for

Reply:
I thought a bit more about this and i think the loop should be instead something like

```
size = 0;
for (i = 0; i < resp->n_numa_fabric_interfaces; i++) {
	/* DAOS_SYS_INFO_STRING_MAX for interface name and 1 for a separator ',' between them */
	size += resp_g->numa_fabric_interfaces[i]->n_ifaces * (DAOS_SYS_INFO_STRING_MAX + 1);
}

D_ALLOC(crt_info->cio_interface, size);
```
otherwise today for each comma you allocate DAOS_SYS_INFO_STRING_MAX
...

Reply:
yea sorry too much multitasking, the +1 was wrong. i mixed up count and size. ill fix this again